src/geometry/euclidean/circle.lean

```diff
@@ -0,0 +1,176 @@
+/-
+Copyright (c) 2021 Manuel Candales. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Manuel Candales
+-/
+import geometry.euclidean.basic
+
+/-!
+# Circles
+
+This file proves basic geometrical results about distances and angles
+in circles in real inner product spaces and Euclidean affine spaces.
+-/
+
+open_locale euclidean_geometry real_inner_product_space real
+
+variables {V : Type*} [inner_product_space ℝ V]
+
+namespace inner_product_geometry
+/-!
+### Geometrical results on circles in real inner product spaces
+
+This section develops some results on circles in real inner product spaces,
+which are used to deduce corresponding results for Euclidean affine spaces.
+-/
+
+lemma mul_norm_eq_sub_norm_pow_two {x y z : V} (h₁ : ∃ k : ℝ, k ≠ 1 ∧ x + y = k • (x - y))
+  (h₂ : ∥z - y∥ = ∥z + y∥) : ∥x - y∥ * ∥x + y∥ = abs (∥z + y∥ ^ 2 - ∥z - x∥ ^ 2) :=
+begin
+  obtain ⟨k, hk_ne_one, hk⟩ := h₁,
+  have hzy : ⟪z, y⟫ = 0,
+  { rw [← eq_of_pow_two_eq_pow_two (norm_nonneg (z - y)) (norm_nonneg (z + y)),
+        norm_add_pow_two_real, norm_sub_pow_two_real] at h₂,
+    linarith },
+
+  let l : ℝ := 1,
```

Comment:
Why not delete this line and just use `1` where applicable? This may break the proof in a couple of spots but it should be easily fixable and should make things simpler in the long run. (If you don't see what I mean by that last part and you can't get it to fix try using `(1 : ℝ)`)

Reply:
I introduced that line, to avoid repeating (1 : ℝ) everywhere. I think it is easier to read that way. Also because of space. I don't like it when a calc line has to be split in two. I rather have shorter lines.

Reply:
Okay. I have an idea for a way that wouldn't require any (1 : ℝ), I'll test it when I get a chance.

Reply:
Tell me your idea. I'll replace by 1's and see how many times (1 : ℝ) is required.

Reply:
It's actually not too bad, just a few places needed (1 : ℝ). I implemented your suggestion.

Reply:
I think if you remove the lines `= (k • x - l • x) - (k • y + l • y) : by rw [sub_smul k l x, add_smul k l y]` and `  ... = ∥r • y - l • y∥ * ∥r • y + l • y∥    : by rw [sub_smul r l y, add_smul r l y]` from the `calc` blocs by combining with the following lines you will not require any (1 : ℝ)s.

Reply:
Looking at your most recent commit, I am now certain that will work.

Reply:
Yeap, I was actually implementing that. Look at my last commit.

Reply:
I would love to eliminate this have statement too: have hl_smul_y_eq_y : (1 : ℝ) • y = y, { rw one_smul },
But I haven't been able to.

Reply:
Ah, nevermind, that can be deleted as well.

Reply:
Try replacing that line of the `calc` block with `by simp_rw [sub_smul, add_smul, one_smul]`

Reply:
The explicit rw works (`rw [sub_smul r 1 y, add_smul r 1 y, one_smul]`)